### PR TITLE
Rqueries

### PR DESCRIPTION
**Pull request by raeyh**
two queries,
*List of Contributor's name for highest rating (>4)
and percentage of unapproved activities

### CODE REVIEW

File: src/main/java/com/example/tour/functional/TourUtilFunctions.java [hunk 1]
```diff
@@ -95,4 +95,22 @@
...
+            .collect(Collectors.toList());
+
+    public static final Function<List<Activity>, Long> countOfNotApprovedActivities = activities -> activities.stream()
+            .filter(r->r.isActive()==false)
+            .count();
+    public static final Function<List<Activity>, Long> countOfAllActivities = activities -> activities.stream()
+            .count();
+    public static final Function<List<Activity>, Double> percentageOfNotApprovedActivities = activities -> (countOfNotApprovedActivities.apply(activities).doubleValue()
+            /countOfAllActivities.apply(activities).doubleValue()) * 100;
+
```
Comment (ivankras) [inline src/main/java/com/example/tour/functional/TourUtilFunctions.java:114]: Maybe getting the percentage of not approved activities from a Place would give us a more complex query (just trying to be in the professor's head for a moment)

File: src/main/java/com/example/tour/functional/TourUtilFunctions.java [hunk 2]
```diff
@@ -95,4 +95,22 @@
...
                 .collect(Collectors.toList());
 
+    /**
+     * --------------------------------------
+     */
+
+    public static final BiFunction<List<Place>, Long, List<String>> starUsers = (places, year) -> activeActivitiesFromPlaces.apply(places).stream()
+            .filter(a -> averageRatingGreaterThanK.test(a, 4D))
+            .map(u->u.getProposedBy().getUsername())
+            .collect(Collectors.toList());
```
Comment (ivankras) [inline src/main/java/com/example/tour/functional/TourUtilFunctions.java:105]: Nice

**Review by ivankras — COMMENTED** [review]
See my comment about the second function.
Also, remember the tests for every function (even the inner ones) and adding a use case to MainFunctional.java :)

Good job

File: src/main/java/com/example/tour/functional/TourUtilFunctions.java (hunk 1 above)
Comment (raeyh) [reply src/main/java/com/example/tour/functional/TourUtilFunctions.java]: yeah you are right,
I changed it,
we can change it if we have better queries

**Pull request closed without merging**